akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -88,12 +130,12 @@ private object TimerMessages {
 abstract class GraphStageLogic private[stream] (inCount: Int, outCount: Int) {
   import GraphInterpreter._
   import TimerMessages._
+  import GraphStageLogic._
 
   def this(shape: Shape) = this(shape.inlets.size, shape.outlets.size)
 
   private val keyToTimers = mutable.Map[Any, Timer]()
   private val timerIdGen = Iterator from 1
```

Comment:
Yes, it is safe, and it can start at any number (it would break if a timer is cancelled and rescheduled four billion times while the message is in flight from the scheduler—which is more than just hypothetical).